From ad5f00c34e26c3b5c7102db9ee1951af3e9bab62 Mon Sep 17 00:00:00 2001
From: fritsch <peter.fruehberger@gmail.com>
Date: Sat, 27 Feb 2016 07:35:28 +0100
Subject: [PATCH] Workaround libdvd (thx milhouse)

---
 Makefile.in | 2 --
 1 file changed, 2 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 8a06e6f..fa1fc03 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -11,7 +11,6 @@ EC_DIRS= \
 
 DVDPCODECS_DIRS= \
 	lib \
-	lib/libdvd 
 
 VideoPlayer_ARCHIVES=xbmc/cores/VideoPlayer/VideoPlayer.a \
                    xbmc/cores/VideoPlayer/DVDCodecs/DVDCodecs.a \
@@ -321,7 +320,6 @@ libaddon: exports
 	$(MAKE) -C lib/addons/library.kodi.guilib
 	$(MAKE) -C lib/addons/library.xbmc.pvr
 dvdpcodecs: dllloader
-	$(MAKE) -C lib/libdvd
 
 dvdpextcodecs:
 
-- 
2.5.0

